urlparse4/cgurl.pyx

```diff
@@ -134,36 +162,45 @@ class SplitResultNamedTuple(tuple):
             elif prop == "fragment":
                 return self[4]
             elif prop == "port":
+                """
+                TODO:
+                Port can go beyond 0
+                """
                 if parsed.port.len > 0:
                     port = int(slice_component(url, parsed.port))
                     if port <= 65535:
                         return port
 
             elif prop == "username":
+                if decoded:
+                    return slice_component(url, parsed.username).decode('utf-8') or None
```

Comment:
would be nice to avoid repeating ``slice_component(url, parsed.username)`` on this line and below